src/elfpy/pricing_models/base.py

```diff
@@ -89,6 +89,11 @@ def model_name(self) -> str:
         """Unique name given to the model, can be based on member variable states"""
         raise NotImplementedError
 
+    @abstractmethod
+    def model_type(self) -> str:
+        """Unique identifier given to the model, should be lower snake_cased name"""
+        raise NotImplementedError
+
```

Comment:
If the model type is always the lower cased version of the name, why don't we just implement this as:

```
return self.model_name().lower()
```